File: 50_Pow(x, n).py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        a=1
        if x>=pow(2,-31) and x<=(pow(2,31)-1):
            if n>0:
                for i in range(n):
                    a=a*x
                return a
            else:
                for i in range(abs(n)):
                    a=a*x
                    
                ans=1/a
                return ans
                
                
            
                
            
                
        else:
            return 0
        
        
            
        

File: 50_Pow(x, n).cpp
class Solution {
public:
    double myPow(double x, int n) {
        
        double result = 1.0;
        
        if (n < 0) {
            x = 1.0 / x;
        }
        
        while(n) {
            
            if (n % 2 != 0) {
                result *= x;
            }
            
            n /= 2;
            
            x *= x;
            
        }
        
        return result;
    }
};